lib/Dialect/SystemC/SystemCOps.cpp

```diff
@@ -291,6 +291,24 @@ LogicalResult SCModuleOp::verifyRegions() {
   return success();
 }
 
+CtorOp SCModuleOp::getOrCreateCtor() {
+  CtorOp ctor;
+  getBody().walk([&](CtorOp op) { ctor = op; });
```

Comment:
This walk may be expensive if things get big. You can prevent some of this by, I think, preventing walking the recursive walk into children with a default hat does WalkResult::skip.